modules/mongodb/charts.go

```diff
@@ -0,0 +1,490 @@
+	Dims: module.Dims{
+		{ID: "catalogStats.collections", Name: "collections"},
+		{ID: "catalogStats.capped", Name: "capped"},
+		{ID: "catalogStats.timeseries", Name: "timeseries"},
+		{ID: "catalogStats.views", Name: "views"},
+		{ID: "catalogStats.internalCollections", Name: "internalCollections"},
+		{ID: "catalogStats.internalViews", Name: "internalViews"},
+	},
+}
+
+var (
+	chartTcmallocGeneric = module.Chart{
+		ID:    "tcmallocGeneric",
+		Title: "Tcmalloc generic metrics",
+		Units: "MiB",
+		Fam:   "tcmalloc",
+		Ctx:   "mongodb.tcmallocGeneric",
+		Dims: module.Dims{
+			{ID: "tcmalloc.generic.current_allocated_bytes", Name: "current_allocated_bytes", Div: 1 << 20},
+			{ID: "tcmalloc.generic.heap_size", Name: "heap_size", Div: 1 << 20},
+		},
+	}
+	chartTcmalloc = module.Chart{
+		ID:    "tcmalloc",
+		Title: "Tcmalloc",
+		Units: "KiB",
+		Fam:   "tcmalloc",
+		Ctx:   "mongodb.tcmalloc",
+		Dims: module.Dims{
+			{ID: "tcmalloc.tcmalloc.pageheap_free_bytes", Name: "Pageheap free", Div: 1024},
+			{ID: "tcmalloc.tcmalloc.pageheap_unmapped_bytes", Name: "Pageheap unmapped ", Div: 1024},
+			{ID: "tcmalloc.tcmalloc.current_total_thread_cache_bytes", Name: "Total threaded cache", Div: 1024},
+			{ID: "tcmalloc.tcmalloc.total_free_bytes", Name: "Free", Div: 1024},
+			{ID: "tcmalloc.tcmalloc.pageheap_committed_bytes", Name: "Pageheap committed", Div: 1024},
+			{ID: "tcmalloc.tcmalloc.pageheap_total_commit_bytes", Name: "Pageheap total commit", Div: 1024},
+			{ID: "tcmalloc.tcmalloc.pageheap_total_decommit_bytes", Name: "Pageheap decommit", Div: 1024},
+			{ID: "tcmalloc.tcmalloc.pageheap_total_reserve_bytes", Name: "Pageheap reserve", Div: 1024},
+		},
+	}
+)
+
+var chartGlobalLockCurrentQueue = module.Chart{
+	ID:    "globalLockCurrentQueue",
+	Title: "Current Queue Clients",
+	Units: "clients",
+	Fam:   "clients",
+	Ctx:   "mongodb.currentQueue",
+	Type:  module.Stacked,
+	Dims: module.Dims{
+		{ID: "globalLock.currentQueue.readers", Name: "readers"},
+		{ID: "globalLock.currentQueue.writers", Name: "writers"},
+	},
+}
+
+var chartMetricsCommands = module.Chart{
+	ID:    "metricsCommand",
+	Title: "Command Metrics",
+	Units: "commands/s",
+	Fam:   "commands",
+	Ctx:   "mongodb.metricsCommand",
+	Dims: module.Dims{
+		{ID: "metrics.commands.eval.total", Name: "Eval", Algo: module.Incremental},
+		{ID: "metrics.commands.eval.failed", Name: "Eval Failed", Algo: module.Incremental},
+		{ID: "metrics.commands.delete.total", Name: "Delete", Algo: module.Incremental},
+		{ID: "metrics.commands.delete.failed", Name: "Delete Failed", Algo: module.Incremental},
+		{ID: "metrics.commands.count.failed", Name: "Count Failed", Algo: module.Incremental},
+		{ID: "metrics.commands.createIndexes", Name: "Create Indexes", Algo: module.Incremental},
+		{ID: "metrics.commands.findAndModify", Name: "Find And Modify", Algo: module.Incremental},
+		{ID: "metrics.commands.insert.failed", Name: "Insert Fail", Algo: module.Incremental},
+	},
+}
```

Comment:
These are [database commands](https://docs.mongodb.com/manual/reference/command/), and there are a lot of commands. Why do we need these and not others?

Also, I think it'd better have 2 charts: executed and failed.

Reply:
We have now removed the command to save on the request size.
Now we do `command := bson.D{{Key: "serverStatus", Value: 1}, {Key: "metrics", Value: 0}, {Key: "repl", Value: 0}}` to half the size of the response